torch/fx/traceback.py

```diff
@@ -55,6 +67,8 @@ def set_current_meta(meta : Dict[str, Any]):
         saved_meta = current_meta
         try:
             current_meta = meta
+            # Record sequence ID right before the node is executed
+            current_meta["seq_id"] = get_sequence_nr()
```

Comment:
This doesn't feel like the right granularity because this entrypoint interposes at the `call_{function},{module},{method}` level, so you are essentially capturing the sequence_nr corresponding to the first node created during that `call_function` - or is that the intended behavior? If it is the intended behavior, how do you plan to use this information downstream?

Reply:
Hi @soulitzer thanks for the code review!  I see what you mean about where to set the seq_nr.  The common case is that setting the seq_nr at the call_{func},{mod},{method} level is sufficient. eg when the ops get lowered to aten they would maintain the same seq_nr.  However,  I found an example where this doesn't hold up.  See the attached profile clip from running in eager mode.  In this case l1_loss has seq_nr 28, but it gets lowered to **aten::abs seq=29** and **aten::mean seq=30**.  Then in the bwd pass **seq=29 and seq=30** are the nodes that have gradients computed and seq=28 doesn't appear in the bwd pass. 

![image](https://github.com/pytorch/pytorch/assets/37422826/4d06efc3-04af-4746-bbb6-92792d67aaad)

Reply:
Ok I added some logic that does a better job at tracking the seq_nr for aten ops that are composed of other aten ops.  In the example above I think aten::l1_loss is treated as a **call_module** but this op is implemented by other aten ops eg aten::abs and aten::mean which are considered inner_ops by the dispatcher logic in torch fx.  I added logic to torch.fx.proxy.create_node() which detects that the op is in the fwd pass and adds updates the seq_nr with the most recent seq_nr captured by autograd.  

Here's an example of how you can use seq_nr for analysis.  I walked the joint fx graph and printed node and node.meta attributes to make this table.  You can see that **seq_nr 755** is the main **matmul** op for the linear layer.  In the bwd pass there are 2 more **aten::mm** ops and 3 **transpose** ops associated with 755.  These correspond to the data and weight gradient computations.  You can also see there is a transpose and squeeze (seq_nr 754 and 755) associated with this same linear op.  They each have a corresponding bwd op you can see at the bottom of the table.

This information is really useful for performance analysis.  The longer term goal with adding sequence_nr to aot_autograd is that I want to be able to observe this in the open ai triton kernels using the profiler.  This information will enable a user to better understand which operators from the original network graph were fused together in the triton kernels.
 
![aot_seq_nr](https://github.com/pytorch/pytorch/assets/37422826/1dbf9ef0-b449-4f77-becf-7f9d70012c80)